test/WebJobs.Script.Tests/NodeEndToEndTests.cs

```diff
@@ -733,6 +733,72 @@ public void ExcludedFunction_NotAddedToHost()
             }
         }
 
+        [Fact]
+        public async Task PromiseApi_Resolves()
+        {
+            JObject input = new JObject
+            {
+                { "scenario", "promiseApiResolves" }
+            };
+
+            Task t = Fixture.Host.CallAsync("Scenarios",
+                new Dictionary<string, object>()
+                {
+                    { "input", input.ToString() }
+                });
+
+            Task result = await Task.WhenAny(t, Task.Delay(5000));
+            Assert.Same(t, result);
+            if (t.IsFaulted)
+            {
+                throw t.Exception;
+            }
+        }
+
+        [Fact]
+        public async Task PromiseApi_Rejects()
+        {
+            JObject input = new JObject
+            {
+                { "scenario", "promiseApiRejects" }
+            };
+
+            Task t = Fixture.Host.CallAsync("Scenarios",
+                new Dictionary<string, object>()
+                {
+                    { "input", input.ToString() }
+                });
+
+            Task result = await Task.WhenAny(t, Task.Delay(5000));
```

Comment:
In a nutshell, the nextTick issue.  If nextTick or Promises start failing the tests will hang, so we need the timeout to shortcircuit & fail the test.